src/server/services/controllers/accountController.js

```diff
@@ -13,47 +14,85 @@ exports.getAllAccounts = function(req, res, next) {
     models.getAllAccounts(function(err, accounts) {
         if (err) {
             sendError(400, res, err, 'Failed on getAllAccounts');
+            return next;
         }
-        res.json(accounts);
+        res.status(200).json(accounts);
     });
 }
 
 exports.getAccountById = function(req, res, next) {
+    if (!req.params && !req.params.id) {
```

Comment:
Shouldn't this be
```javascript
if (!req.params || !req.params.id)
```
?
Otherwise, in the case where `req.params` is an object, even an empty object, but `req.params.id` is `undefined` or `null`, `(!req.params && !req.params.id)` evaluates to false and this block won't execute.